docs/overview/04-proof-aggregation/03-publish-attestations.md

```diff
@@ -0,0 +1,28 @@
+---
+title: Publish Attestations
+---
+
+Every time an aggregation is complete a [`AggregationComplete{domainId, aggregationId}`](../02-mainchain/05-mainchain_api.md#aggregationcomplete) event is emitted; the aggregation now is complete, no more proofs can be added to it and is moved in the publishing queue.
+
+In order to generate an aggregation receipt (the Merkle root) it's sufficient that any user calls [`aggregate(domainId, aggregationId)`](../02-mainchain/05-mainchain_api.md#aggregate_3) with the correct coordinates. Build this aggregation has a cost because you should pay the effort to generate the tree, compute the root and emit the events, but the user will receive the money that was collected by the [proof submitters](./02-handle-valid-proof.md#proof-submission-costs). The publisher should take care that the estimations did to hold the money from the proof submitters can be different from the actual price: in the normal flow a publisher should evaluate if it's worth of generate the aggregation by inspect the storage before doing the call.
```

Comment:
```suggestion
In order to generate an aggregation receipt (the Merkle root) it's sufficient that any user calls [`aggregate(domainId, aggregationId)`](../02-mainchain/05-mainchain_api.md#aggregate_3) with the correct coordinates. Building this aggregation has a cost because you should pay for the computation to generate the tree, compute the root and emit the events, but the user will receive the money that was collected by the [proof submitters](./02-handle-valid-proof.md#proof-submission-costs). The publisher should take into account the fact that the estimations he did to hold the money from the proof submitters can be different from the actual price: in the normal flow a publisher should evaluate if it's worth to generate the aggregation by inspecting the storage before doing the call.
```